sopel/config/core_section.py

```diff
@@ -96,8 +96,23 @@ class CoreSection(StaticSection):
     channels = ListAttribute('channels')
     """List of channels for the bot to join when it connects"""
 
+    db_type = ValidatedAttribute('db_type')
+    """The type of database to use for Sopel's database. (SQLite or MySQL)"""
```

Comment:
Can we simply refer to sqlalchemy docs instead of limiting choices?

Reply:
That's probably going to be infrequent though and a minor PR to address, that seems like a reasonable choice. Downside to this is that it limits usability in non-standard databases. There are some that ship supported in SQLAlchemy and others that are through external packages.

The docs for what Dialects (databases) SQLAlchemy supports (as in listed on their website): https://docs.sqlalchemy.org/en/latest/dialects/index.html

After thinking about it, given that SQLAlchemy could have any database plugin written and then consumed since anyone could write a new Dialect for SQLAlchemy, it'd probably be best to not limit this by using a ChoiceAttribute. I think warning when a connection couldn't be made instead is best.

Reply:
I've made this a ChoiceAttribute for now. It looks like there aren't _that_ many dialects to deal with (7), and honestly people only probably use 4 of those (sqlite, mysql, postgres, and mssql).

The biggest question is how we are going to handle dependencies for these, since mysql requires pymysql, etc. Is that something that should be documented or should the PyMySQL dependency be included in requirements (which would also need to include the dependencies for _all_ other database dialects, and that feels bad)

Reply:
Dependencies for the various dialects should be left to documentation, IMO. Adding them all to requirements would partially defeat the purpose of making the backend a choice, because the point is to use SQLite (from Python's standard library) unless the user specifically sets it up otherwise. And to do that, they should need to read some documentation. :smile_cat: